### PR TITLE
Update documentation. Extend tests in travis.yaml

### PR DESCRIPTION
**Pull request by baor**
With this PR
* Description for shared objects is updated
* README is extended with a use-case
* travis.yaml is updated to do benchmarks

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -435,6 +435,29 @@ func (m Message) LogErr(err error) {
 	}
 }
 
+// msg marshals the message from template and sends it to output stream
+// msg - an argument for additional text message
+func (m Message) msg(msg string) {
```
Comment (Prutswonder) [inline logger.go]: I don't think it's a good idea to use the same name for the parameter the and the method. In this case it's more clear to rename the method to `send()`, which is more in line with it's intent (performing an action).

Comment (baor) [reply logger.go]: what you think name "print"? 
Because I am not really sending it...

Comment (Prutswonder) [reply logger.go]: But you're using the word "send" in the comment 😄 

If it's not really sending, then maybe you can leave the method name as-is, but instead rename the parameter "msg" to "message" or "msgToSend".

Comment (baor) [reply logger.go]: I will rename it to text :)
thanks!

**Pull request merged**